jwst/skymatch/tests/test_skymatch.py

```diff
@@ -515,7 +517,7 @@ def test_skymatch_2x(tmp_cwd, nircam_rate, tmp_path, skymethod, subtract):
 
     # 2nd run.
     step.subtract = subtract
-    result2 = step.run(result)
+    result2 = step.run(asn_out_fname)
```

Comment:
```suggestion
    result2 = step.run(result)
```
I think this test expects that skymatch is running on results from a previous skymatch run (not on the same input a second time). Did running with `result` fail?

Reply:
It was raising a `BorrowError: Attempt to double-borrow model`, but I didn't fix it in the right way, so thanks for catching this.  It appears that the lines above that say
```
with pytest.raises(ValueError):
    step.subtract = not subtract
    step.run(result)
```
lead to this BorrowError.  The `ValueError` that is raised is [this one](https://github.com/spacetelescope/jwst/blob/1a45e3480c6e7d6c103a188c04ce4041084fc644/jwst/skymatch/skymatch_step.py#L195), which happens when a model is open.

I verified that if these lines are removed, `step.run()` can be called an arbitrary number of times on the previous iteration's input without any failures.  But removing those lines is not necessarily a solution... what is the recommended way to handle this with a ModelLibrary?  Do I need to put explicit calls to `shelve` right before the `raise` statements?

Reply:
I tested this, and putting an explicit shelve before the raise does work.  Not sure there's any way to avoid this?

Reply:
Thanks for testing that. Shelving before the raise is the only way to avoid the borrow error that I can think of.

Part of me thinks the test is doing too much but that's probably scope creep. So maybe just a `try finally` block wrapping `_imodel2skyim` with a `shelve` in the `finally` is good enough.